### PR TITLE
Add BuildKit Task

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Signed-off-by: Akihiro Suda <akihiro.suda.cz@hco.ntt.co.jp>

# Changes

Add BuildKit Task

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

cc @tonistiigi @tiborvass

Push is currently untested due to https://github.com/tektoncd/pipeline/issues/848

### CODE REVIEW

**Comment by tiborvass** [comment]
Ping @vdemeester

**Comment by AkihiroSuda** [comment]
hmm,  `buildctl` fails with `error: rpc error: code = Unknown desc = failed to fetch anonymous token: unexpected status: 401 Unauthorized`...

The credential was created with  https://github.com/knative/build-templates/blob/fa2ac1bc3f31d6d7a367e1c783c116b1f6b94ce0/gcr_helper/helper.sh but `s/build.knative.dev/tekton.dev/g`, and this credential works fine with kaniko and buildah.

Note that  `buildctl` with `docker-credential-gcloud` helper (out of cluster) works fine 

cc @tonistiigi

**Comment by AkihiroSuda** [comment]
Confirmed that this task can push the image to `gcr.io` successfully

File: buildkit/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# BuildKit
+
+This Task builds source into a container image using [Moby BuildKit](https://github.com/moby/buildkit).
+
+Rootless mode is used by default.
+
+## Parameters
```
Comment (ImJasonH) [inline buildkit/README.md]: In the other catalog tasks, we have `## Inputs` and `### Parameters` and `### Resources`, and then `## Outputs` / `### Resources` -- can you do that here too?

Comment (AkihiroSuda) [reply buildkit/README.md]: updated

**Comment by ImJasonH** [comment]
/approve
/lgtm

:rocket:

**Pull request merged**